### PR TITLE
ROC-3144: redirect to error page for claim over 10k

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3144

### Change description
Display error page when claim amount + interest > £10k

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [x] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [x] UI changes look good on mobile
- [x] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/app/fees/feesClient.ts [hunk 1]
```diff
@@ -22,9 +22,8 @@ export class FeesClient {
    * @param {number} claimValue the amount claiming for in pounds
    * @returns {Promise.<number>} promise containing the fee amount in pounds
    */
-  static calculateIssueFee (claimValue: number): Promise<number> {
-    return this.calculateFee(issueFeeCode, claimValue)
-      .then((outcome: CalculationOutcome) => MoneyConverter.convertPenniesToPounds(outcome.amount))
+  static async calculateIssueFee (claimValue: number): Promise<number> {
+    return MoneyConverter.convertPenniesToPounds(((await this.calculateFee(issueFeeCode, claimValue))).amount)
```
Comment (timja-kainos) [inline src/main/app/fees/feesClient.ts]: can you extract to a variable the fee calculation this is quite hard to read?
REC

Comment (geek-soft) [reply src/main/app/fees/feesClient.ts]: I agree

File: src/main/app/fees/feesClient.ts [hunk 2]
```diff
@@ -45,39 +43,37 @@ export class FeesClient {
...
-  private static getRangeGroup (code: string): Promise<RangeGroup> {
+  private static async getRangeGroup (code: string): Promise<RangeGroup> {
     if (StringUtils.isBlank(code)) {
       throw new Error('Fee code is required')
     }
-    return request.get(`${feesUrl}/range-groups/${code}`)
-      .then((body: Object) => {
-        return plainToClass(RangeGroup, body)
-      })
+    return plainToClass(RangeGroup, await request.get(`${feesUrl}/range-groups/${code}`) as object)
```
Comment (geek-soft) [inline src/main/app/fees/feesClient.ts]: I would recommend to extract response as local variable.

File: src/main/features/claim/routes/fees.ts [hunk 3]
```diff
@@ -8,43 +8,52 @@ import { RangeGroup } from 'app/fees/models/rangeGroup'
...
-        FeesClient.calculateIssueFee(claimAmount),
-        FeesClient.calculateHearingFee(claimAmount),
-        FeesClient.getIssueFeeRangeGroup(),
-        FeesClient.getHearingFeeRangeGroup()
-      ]
-    )
-      .then((values: any[]) => {
-        const issueFeeRangeGroup: RangeGroup = values[2]
-        const hearingFeeRangeGroup: RangeGroup = values[3]
+  .get(Paths.feesPage.uri,
```
Comment (geek-soft) [inline src/main/features/claim/routes/fees.ts]: Please add route test that reflects scenario you are fixing. In test case draft should have an amount set to 10k and interest declared from some date in the past that will result in final amount being over 10k.

REQ

File: src/main/features/claim/routes/fees.ts [hunk 4]
```diff
@@ -8,43 +8,52 @@ import { RangeGroup } from 'app/fees/models/rangeGroup'
 import { FeesTableViewHelper } from 'claim/helpers/feesTableViewHelper'
 import { DraftClaim } from 'drafts/models/draftClaim'
 import { Draft } from '@hmcts/draft-store-client'
+import { ClaimAmountExceedsLimitError } from '../../../errors'
```
Comment (rlewan) [inline src/main/features/claim/routes/fees.ts]: Can you please change this to not be a relative import?

File: src/main/app/fees/feesClient.ts (hunk 1 above)
Comment (yarpo) [reply src/main/app/fees/feesClient.ts]: done

File: src/main/app/fees/feesClient.ts (hunk 2 above)
Comment (yarpo) [reply src/main/app/fees/feesClient.ts]: done

File: src/main/features/claim/routes/fees.ts (hunk 3 above)
Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: done

File: src/main/features/claim/routes/fees.ts [hunk 5]
```diff
@@ -8,43 +8,52 @@ import { RangeGroup } from 'app/fees/models/rangeGroup'
...
-          .map(range => range.copy({ to: Math.min(range.to, supportedFeeLimitInPennies) }))
+      Promise.all(
+        [
+          FeesClient.calculateIssueFee(claimAmount),
+          FeesClient.calculateHearingFee(claimAmount),
+          FeesClient.getIssueFeeRangeGroup(),
+          FeesClient.getHearingFeeRangeGroup()
+        ]
+      )
+        .then((values: any[]) => {
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: It't not my code :)

I only pressed "reformat".

File: src/main/features/claim/routes/fees.ts [hunk 6]
```diff
@@ -16,6 +16,7 @@ export default express.Router()
   .get(Paths.feesPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
     const draft: Draft<DraftClaim> = res.locals.claimDraft
     const claimAmount: number = await claimAmountWithInterest(draft.document)
+
```
Comment (geek-soft) [inline src/main/features/claim/routes/fees.ts]: I think it would be good to revert that change as it does not add anything.

Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: done

File: src/main/features/claim/routes/fees.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: done

File: src/main/features/claim/routes/fees.ts [hunk 7]
```diff
@@ -8,43 +8,52 @@ import { RangeGroup } from 'app/fees/models/rangeGroup'
...
+          res.render(Paths.feesPage.associatedView,
+            {
+              issueFee: values[0],
+              hearingFee: values[1],
+              rows: FeesTableViewHelper.merge(supportedIssueFees, supportedHearingFees)
+            }
+          )
+        })
+        .catch((error) => {
+          if (error instanceof ClaimAmountExceedsLimitError) {
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: done

File: src/main/features/claim/routes/fees.ts (hunk 4 above)
Comment (yarpo) [reply src/main/features/claim/routes/fees.ts]: fixed

**Pull request merged**